src/config.rs

```diff
@@ -68,6 +74,98 @@ impl Config {
     }
 }
 
+#[cfg(unix)]
+type Child = std::process::Child;
+#[cfg(windows)]
+type Child = create_process_w::Child;
+
+#[cfg(unix)]
+#[derive(Serialize, Deserialize)]
+pub struct SubProcess {
+    pub command: String,
+    pub foreground: bool,
+    #[serde(default)]
+    pub keep_on_exit: bool,
+    pub working_dir: Option<PathBuf>,
+    pub stdout: Option<bool>,
+    pub stderr: Option<bool>,
+}
+#[cfg(windows)]
+#[derive(Serialize, Deserialize)]
+pub struct SubProcess {
+    pub command: String,
+    pub foreground: bool,
+    #[serde(default)]
+    pub keep_on_exit: bool,
+    pub working_dir: Option<PathBuf>,
+    pub inherit_handles: Option<bool>,
+}
+
+impl SubProcess {
+    pub fn spawn(&self, tmp_dir: &Path) -> Option<Child> {
+        let mut process = {
+            #[cfg(unix)]
+            {
+                let mut process = std::process::Command::new(run::get_shell());
+                process
+                    .current_dir(self.working_dir.as_deref().unwrap_or(tmp_dir))
+                    .args(["-c", &self.command])
+                    .stdin(std::process::Stdio::null());
+
+                if !self.foreground {
+                    if !self.stdout.unwrap_or(false) {
+                        process.stdout(std::process::Stdio::null());
+                    }
+
+                    if !self.stderr.unwrap_or(false) {
+                        process.stderr(std::process::Stdio::null());
+                    }
+                } else {
+                    if !self.stdout.unwrap_or(true) {
+                        process.stdout(std::process::Stdio::null());
+                    }
+
+                    if !self.stderr.unwrap_or(true) {
+                        process.stderr(std::process::Stdio::null());
+                    }
+                }
+
+                process
+            }
+            #[cfg(windows)]
+            {
+                let mut process = create_process_w::Command::new(&self.command);
+                process
+                    .current_dir(self.working_dir.as_deref().unwrap_or(tmp_dir))
+                    .inherit_handles(self.inherit_handles.unwrap_or(false));
```

Comment:
`false` is not necessarily the default of this library. It is but since it is third party you must not assume things.

Set the inherit_handles only if it is Some. Otherwise don't set anything.